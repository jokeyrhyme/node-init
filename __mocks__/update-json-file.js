/* @flow */
'use strict';

let pkg /* : Object */ = {};

module.exports = async (
  pkgPath /* : string */,
  updater /* : (value: any) => Promise<any> */,
) /* : Promise<void> */ => {
  await updater(pkg);
};

module.exports.__setPackage = (p /* : Object */) => {
  pkg = p;
};
