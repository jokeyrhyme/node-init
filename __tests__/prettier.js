'use strict';

jest.mock('update-json-file');

const os = require('os');

const execa = require('execa');

const {
  lib: { npmScript },
} = require('../lib/tasks/prettier.js');

const TEN_SECONDS = 10 * 1e3;
jest.setTimeout(TEN_SECONDS);

test('`npm run prettier` finds correct files', async () => {
  let { stdout } = await execa('npm', [
    'run',
    'prettier',
    '--',
    '--no-list-different',
  ]);

  // output on Windows includes the full path to prettier
  // so we have to strip it before the next assertion
  if (/^win/.test(os.platform())) {
    stdout = stdout.replace('\\node_modules\\prettier\\', '\\');
  }

  expect(stdout).not.toMatch(/flow-typed/);
  expect(stdout).not.toMatch(/node_modules/);
});

test('npmScript() versus empty package.json', async () => {
  const pkg = {};
  require('update-json-file').__setPackage(pkg);

  await npmScript('some/path');

  expect(pkg).toMatchSnapshot();
});

test('npmScript() versus engines.node >= 8', async () => {
  const pkg = {
    engines: { node: '>=8' },
  };
  require('update-json-file').__setPackage(pkg);

  await npmScript('some/path');

  expect(pkg).toMatchSnapshot();
});
