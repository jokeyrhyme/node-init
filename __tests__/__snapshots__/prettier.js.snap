// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`npmScript() versus empty package.json 1`] = `
Object {
  "prettier": Object {
    "singleQuote": true,
    "trailingComma": "es5",
  },
  "scripts": Object {
    "pretest": "npm run prettier",
    "prettier": "npx -q prettier --list-different --write \\"**/*.{css,js,jsx,less,scss,ts,tsx}\\" || exit 0",
  },
}
`;

exports[`npmScript() versus engines.node >= 8 1`] = `
Object {
  "engines": Object {
    "node": ">=8",
  },
  "prettier": Object {
    "singleQuote": true,
    "trailingComma": "all",
  },
  "scripts": Object {
    "pretest": "npm run prettier",
    "prettier": "npx -q prettier --list-different --write \\"**/*.{css,js,jsx,less,scss,ts,tsx}\\" || exit 0",
  },
}
`;
